### ISSUE TITLE
nutanix_clusters data source makes Terraform crash

### ISSUE THREAD

**Issue by JonKohler** [nutanix/terraform-provider-nutanix#10]
from https://github.com/nutanix/terraform-provider-nutanix/issues/215 @pipoe2h

**Describe the bug**
A simple Terraform file with the following code for nutanix_cluster data sources makes crash Terraform:

`data "nutanix_clusters" "clusters" {
}`

`terraform apply        122ms  Thu 25 Oct 19:36:13 2018
data.nutanix_clusters.clusters: Refreshing state...

Error: Error refreshing state: 1 error(s) occurred:

* data.nutanix_clusters.clusters: 1 error(s) occurred:

* data.nutanix_clusters.clusters: data.nutanix_clusters.clusters: unexpected EOF`

**Expected behavior**
This should not make crash Terraform.

**Logs**
https://gist.github.com/pipoe2h/203f2324a1f20f39b1309ea9aac0112d

**Versions (please complete the following information):**
 - macOS High Sierra 10.13.6
 - Terraform v0.11.10 
 - Nutanix Cluster (5.9)
 - Nutanix Prism Central (5.9)
 - Terraform provider version (1.0.0)

**Additional context**
Terraform documentation for Nutanix provider is not clear for some of the data sources. For example nutanix_clusters is referencing to images. It seems copy&paste has been used.

https://www.terraform.io/docs/providers/nutanix/d/clusters.html

**Comment by sandeepmp** [comment]
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by podwhitehawk** [comment]
I'm experiencing same behavior when Terraform crashing when using `data "nutanix_clusters" "clusters" { }`.
Tested against:
Nutanix Version 5.8.2 - Ultimate License
NCC Version: 3.6.1
LCM Version: 2.0.2257
Terraform provider version (1.0.0)
I've tried all available Terraform versions from 0.11.8 to 0.11.11.
Also I've tried to use it with 0.12.0-alpha4, but it's totally unusable with error "incompatible API client v4, v5 needed."

here is a full output:
```panic: runtime error: invalid memory address or nil pointer dereference
2019-01-08T11:12:10.308-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: [signal 0xc0000005 code=0x0 addr=0x0 pc=0xc2e32a]
2019-01-08T11:12:10.308-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:
2019-01-08T11:12:10.308-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: goroutine 73 [running]:
2019-01-08T11:12:10.308-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: github.com/terraform-providers/terraform-provider-nutanix/nutanix.dataSourceNutanixClustersRead(0xc0423f6380, 0xcdad20, 0xc0423fc338, 0xc0423f6380, 0x0)
2019-01-08T11:12:10.308-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-nutanix/nutanix/data_source_nutanix_clusters.go:665 +0xeea
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: github.com/terraform-providers/terraform-provider-nutanix/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).ReadDataApply(0xc0421f1a40, 0xc0421827a0, 0xcdad20, 0xc0423fc338, 0xc0425464b0, 0x4d1101, 0xcdf4a0)
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-nutanix/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:292 +0x8f
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: github.com/terraform-providers/terraform-provider-nutanix/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).ReadDataApply(0xc042533c00, 0xc04217a140, 0xc0421827a0, 0x18, 0x18, 0xc042576da0)
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-nutanix/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:426 +0xa1
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: github.com/terraform-providers/terraform-provider-nutanix/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).ReadDataApply(0xc0421ea460, 0xc0423f2b00, 0xc0423f2c70, 0x0, 0x0)
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-nutanix/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:604 +0x55
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: reflect.Value.call(0xc0423866c0, 0xc0423fdbb8, 0x13, 0xe8baa0, 0x4, 0xc042483f18, 0x3, 0x3, 0xc042081e40, 0xc042063000, ...)
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/goenv/versions/1.10.3/src/reflect/value.go:447 +0x970
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: reflect.Value.Call(0xc0423866c0, 0xc0423fdbb8, 0x13, 0xc042483f18, 0x3, 0x3, 0xc0421ea460, 0x0, 0x0)
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/goenv/versions/1.10.3/src/reflect/value.go:308 +0xab
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: net/rpc.(*service).call(0xc0423ef080, 0xc0424c8a50, 0xc0424da580, 0xc0424da590, 0xc042446e80, 0xc0425768e0, 0xcdf460, 0xc0423f2b00, 0x16, 0xcdf4a0, ...)
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/goenv/versions/1.10.3/src/net/rpc/server.go:384 +0x155
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe: created by net/rpc.(*Server).ServeCodec
2019-01-08T11:12:10.309-0600 [DEBUG] plugin.terraform-provider-nutanix_v1.0.0_x4.exe:   /opt/goenv/versions/1.10.3/src/net/rpc/server.go:480 +0x441
2019/01/08 11:12:10 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.nutanix_image.heis: unexpected EOF
2019/01/08 11:12:10 [ERROR] root: eval: *terraform.EvalSequence, err: data.nutanix_image.heis: unexpected EOF
2019/01/08 11:12:10 [TRACE] [walkRefresh] Exiting eval tree: data.nutanix_image.heis
2019/01/08 11:12:10 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.nutanix_clusters.clusters: unexpected EOF
2019/01/08 11:12:10 [ERROR] root: eval: *terraform.EvalSequence, err: data.nutanix_clusters.clusters: unexpected EOF
2019/01/08 11:12:10 [TRACE] [walkRefresh] Exiting eval tree: data.nutanix_clusters.clusters
2019/01/08 11:12:10 [TRACE] dag/walk: upstream errored, not walking "provider.nutanix (close)"
2019/01/08 11:12:10 [DEBUG] plugin: waiting for all plugin processes to complete...
2019-01-08T11:12:10.312-0600 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2019-01-08T11:12:10.322-0600 [DEBUG] plugin: plugin process exited: path=C:\Users\oshkurko\nutanix-terraform\.terraform\plugins\windows_amd64\terraform-provider-nutanix_v1.0.0_x4.exe




!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!```

**Comment by sandeepmp** [comment]
This is working on Prism Central 5.9.1 version.
However fails for Prism Central 5.10.1

**Comment by fuhbar** [comment]
Same here, fails on Prism Central 5.10.1 ☹️

**Comment by marinsalinas** [comment]
Hello, I opened a PR #38  fixing this.

**Comment by maxaudron** [comment]
We are also experiencing a similar error but caused by another part of the code:
https://gist.github.com/3f83845a8c35dcbba06e597283e04ebd

@marinsalinas i tried this with your fix and it still errors.

**Comment by marinsalinas** [comment]
Hello @maxaudron,  I've fixed this on #39, Can you take a look there?

**Comment by maxaudron** [comment]
@marinsalinas yes this fixed it. Thank you very much :)
